File: Python Equivalent/Ceaser.py
def encrypt(text, key):
    key %= 26
    result = ""
    for ch in text:
        if ch.isupper():
            result += chr((ord(ch) - ord('A') + key) % 26 + ord('A'))
        elif ch.islower():
            result += chr((ord(ch) - ord('a') + key) % 26 + ord('a'))
        else:
            result += ch
    return result


def decrypt(text, key):
    key %= 26
    result = ""
    for ch in text:
        if ch.isupper():
            result += chr((ord(ch) - ord('A') - key) % 26 + ord('A'))
        elif ch.islower():
            result += chr((ord(ch) - ord('a') - key) % 26 + ord('a'))
        else:
            result += ch
    return result


text = input("Enter any string: ")
key = int(input("Enter the key: "))

encrypted = encrypt(text, key)
print("\nEncrypted String is:", encrypted)

decrypted = decrypt(encrypted, key)
print("Decrypted String is:", decrypted)

File: CNS Lab (Sem 7)/Ceaser.m
str = input('Enter any string: ', 's'); 
key = input('Enter the Key: ');

encrypted = encrypt(str, key); 
fprintf('\nEncrypted String is: %s\n', encrypted);

decrypted = decrypt(encrypted, key); 
fprintf('Decrypted String is: %s\n', decrypted);


function encrypted = encrypt(str, key) 
    encrypted = '';
    key = mod(key, 26);
    
    for i = 1:length(str) 
        ch = str(i);
        
        if isstrprop(ch, 'upper') 
            c = double(ch) + key; 
            if c > double('Z')
                c = c - 26;
            end
        elseif isstrprop(ch, 'lower') 
            c = double(ch) + key;
            if c > double('z')
                c = c - 26;
            end 
        else
            c = double(ch); 
        end
        
        % Append the character to the result string
        encrypted(end + 1) = char(c); 
    end
end

function decrypted = decrypt(str, key) 
    decrypted = '';
    key = mod(key, 26);
    
    for i = 1:length(str) 
        ch = str(i);
        
        if isstrprop(ch, 'upper') 
            c = double(ch) - key; 
            if c < double('A')
                c = c + 26;
            end
        elseif isstrprop(ch, 'lower') 
            c = double(ch) - key;
            if c < double('a') 
                c = c + 26;
            end 
        else
            c = double(ch);
        end
        
        % Append the character to the result string
        decrypted(end + 1) = char(c); 
    end
end  
